src/zenml/utils/env_utils.py

```diff
@@ -152,3 +153,36 @@ def _substitution_func(v: str) -> str:
     return string_utils.substitute_string(
         value=value, substitution_func=_substitution_func
     )
+
+
+@contextlib.contextmanager
+def temporary_environment(environment: Dict[str, str]) -> Iterator[None]:
```

Comment:
This temporary context will only affect the current process. I think users may want to set pipeline-level environment variables that will have a global effect in the environment where the pipeline steps are running. That means that these variables should be injected somehow into `orchestrator.prepare_or_run_pipeline` and `step_operator.launch`. Is this possible ?

Reply:
@stefannica That was actually my initial intention for this PR, however there are some complications: `orchestrator.prepare_or_run_pipeline` currently accepts a dictionary of environment variables. To make this work with our per-step configurations, we'd have to either change the public API of the orchestrator and pass different environments for all steps. Or we could expect/make it a convention that each orchestrator should update this dict with the step-specific environment configuration that is defined in the config. Or I guess we could make this a pipeline-only configuration, and don't allow setting specific environment variables for a steps, but I'm not a huge fan of this to be honest.

Reply:
WDYT, maybe the `convention` solution is the easiest way to go?
The orchestrator implementations would look something like this:
```
def prepare_or_run_pipeline(...):
  for step in deployment.steps():
    full_env = { **environment, **step.config.environment}
    ...
```

Reply:
This can work, yeah :smiling_face_with_tear:

Reply:
Alright, I'll adjust it then. Our local orchestrator currently ignores these environment variables. Do you think we should also set them for consistency? If yes, does that also include the auth ones that are coming from `prepare_or_run_pipeline`?